### ISSUE TITLE
NoSuchMethodError on RedisLockProvider

### ISSUE THREAD

**Issue by marcioviegas** [lukas-krecan/ShedLock#52]
After some tests scaling my application I start receiving this error

`   2018-02-15T11:15:00.00-0200 [APP/PROC/WEB/0] OUT java.lang.NoSuchMethodError: org.springframework.data.redis.core.types.Expiration.from(Ljava/time/Duration;)Lorg/springframework/data/redis/core/types/Expiration;
   2018-02-15T11:15:00.00-0200 [APP/PROC/WEB/0] OUT 	at net.javacrumbs.shedlock.provider.redis.spring.RedisLockProvider.getExpiration(RedisLockProvider.java:72) ~[shedlock-provider-redis-spring-0.18.0.jar!/:na]
   2018-02-15T11:15:00.00-0200 [APP/PROC/WEB/0] OUT 	at net.javacrumbs.shedlock.provider.redis.spring.RedisLockProvider.lock(RedisLockProvider.java:60) ~[shedlock-provider-redis-spring-0.18.0.jar!/:na]
   2018-02-15T11:15:00.00-0200 [APP/PROC/WEB/0] OUT 	at net.javacrumbs.shedlock.core.DefaultLockManager.executeWithLock(DefaultLockManager.java:51) ~[shedlock-core-0.18.0.jar!/:na]
   2018-02-15T11:15:00.00-0200 [APP/PROC/WEB/0] OUT 	at net.javacrumbs.shedlock.core.DefaultLockingTaskExecutor.executeWithLock(DefaultLockingTaskExecutor.java:38) ~[shedlock-core-0.18.0.jar!/:na]
   2018-02-15T11:15:00.00-0200 [APP/PROC/WEB/0] OUT 	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-4.3.9.RELEASE.jar!/:4.3.9.RELEASE]`

**Comment by lukas-krecan** [comment]
Hi, it seems you have older version of spring-data-redis. Will try to make it compatible with older versions in next few days, in the meantime you can try to update.

**Comment by marcioviegas** [comment]
I'm using spring-data-redis:1.8.4, which one should I be using instead? 

PS: that version was added from spring-boot-starter-data-reds:1.5.10.RELEASE (the latest one from spring boot)

**Comment by clamey** [comment]
The pom.xml for shedlock-provnider-redis-spring has this dependency:

```
        <dependency>
            <groupId>org.springframework.data</groupId>
            <artifactId>spring-data-redis</artifactId>
            <version>2.0.3.RELEASE</version>
        </dependency>
```

**Comment by clamey** [comment]
Looks like the "from" using Duration is new, but this one has been in spring-data-redis since 1.7:

```
Expiration.from(long expirationTime, TimeUnit timeUnit)
```

**Comment by lukas-krecan** [comment]
Yes, I know. I have used Expiration.from from spring-redis 2

**Comment by lukas-krecan** [comment]
Released as 1.18.1
